crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1576,9 +1566,15 @@ impl RoomInfo {
         self.room_state
     }
 
-    /// Returns whether this is an encrypted room.
-    pub fn is_encrypted(&self) -> bool {
-        self.base_info.encryption.is_some()
+    /// Returns the encryption state of this room.
+    pub fn encryption_state(&self) -> EncryptionState {
+        if !self.encryption_state_synced {
+            EncryptionState::Unknown
+        } else if self.base_info.encryption.is_some() {
+            EncryptionState::Encrypted
+        } else {
+            EncryptionState::NotEncrypted
+        }
```

Comment:
It would probably be nice to replace `encryption_state_synced` and `base_info.encryption`with `EncryptionState` where `Encrypted` keeps the `m.room.encryption` content as associated data.

But that's probably something for a different PR and might not even be worth the trouble.

Reply:
Talk about that in a meeting and agreed to address that in another PR.